### PR TITLE
Transportation management (add, delete, etc.)

### PR DESCRIPTION
**Pull request by a161527**
…ceID, a flightID, both or neither (lists all)), add, delete, edit, assignFlight

### CODE REVIEW

**Comment by brandonson** [comment]
I think you've got some stray tabs in there that are messing up your indentation.

Other than that looks good to me.

**Comment by nikkimartin** [comment]
@brandonson and @a161527 - can we discuss this over slack again? I'm still not sure why we're ditching the user association

**Comment by brandonson** [comment]
I'm not really concerned either way - if having users attached is easier I think we might as well do it that way.

**Comment by nikkimartin** [comment]
I think it would make the most sense in terms of tracking capacity and groupings correctly. Also because it doesn't make sense to me not to have reusable transportation records. @movingpylon do you have any thoughts

**Comment by nikkimartin** [comment]
talked w/ @movingpylon, can you change it back to the old schema?

thanks.

**Comment by brandonson** [comment]
To clarify: we want transport to be one table (transportation) and then connected to conference + user via some other table (user_transportation)? Is that right?

Or something like that, table names aren't super important.

**Comment by nikkimartin** [comment]
yup, sounds correct. so you should be able to associate one transportation (i.e. the taxi driver or what have you) w/ several users

File: app/Http/Controllers/TransportationController.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    private function insertTransport($data)
+    {
+        $transport = new Transportation();
+        $transport->capacity = $data['capacity'];
+        $transport->name = $data['name'];
+        $transport->company = $data['company'];
+        $transport->phone = $data['phone'];
+        $transport->conferenceID = $data['conferenceID'];
+        if (isset($data['flightID']))
+            $transport->flightID = $data['flightID'];
```
Comment (nikkimartin) [inline app/Http/Controllers/TransportationController.php]: no flight in transport table

File: app/Http/Controllers/TransportationController.php [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    public function __construct()
+    {
+        // $this->middleware('jwt.auth', ['except' => ['authenticate', 'token']]);
+    }
+    private function transportValidator(array $data)
+    {
+        $validator = Validator::make($data, [
+            '*.capacity'        =>    'required|numeric|min:1|max:100',
+            '*.name'            =>    'required|max:255',
+            '*.company'            =>    'required|max:255',
```
Comment (nikkimartin) [inline app/Http/Controllers/TransportationController.php]: company is optional

File: app/Http/Controllers/TransportationController.php [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    }
+
+    private function insertTransport($data)
+    {
+        $transport = new Transportation();
+        $transport->capacity = $data['capacity'];
+        $transport->name = $data['name'];
+        $transport->company = $data['company'];
+        $transport->phone = $data['phone'];
+        $transport->conferenceID = $data['conferenceID'];
```
Comment (nikkimartin) [inline app/Http/Controllers/TransportationController.php]: add the conferenceID fk into the transportation field

**Comment by nikkimartin** [comment]
sorry, i accidentally commented on the outdated diff w/ a couple more comments. since we're tying the individual transport records w/ conference, could you also set up the urls to match the ones like inventory and residence (i.e. /api/conferences/{cid}/transportation)

thanks!

**Comment by a161527** [comment]
Updated

**Comment by nikkimartin** [comment]
got this error back when i tried inserting w/ out company field:

{message: "unknown_error", errors: "Undefined index: company"}

**Pull request merged**